pkg/morph/event/listener.go

```diff
@@ -284,17 +349,72 @@ func (l listener) parseAndHandleNotification(notifyEvent *state.NotificationEven
 	}
 }
 
+func (l listener) parseAndHandleNotary(nr *response.NotaryRequestEvent) {
+	// prepare the notary event
+	notaryEvent, err := l.notaryEventsPreparator.Prepare(nr.NotaryRequest)
+	if err != nil {
+		if !errors.Is(err, ErrTXAlreadyHandled) {
+			l.log.Warn("could not prepare notary event",
+				zap.String("error", err.Error()),
+			)
+		}
+
+		return
+	}
+
+	notaryKey := notaryRequestTypes{}
+	notaryKey.SetMempoolType(nr.Type)
+	notaryKey.SetRequestType(notaryEvent.Type())
+	notaryKey.SetScriptHash(notaryEvent.ScriptHash())
+
+	// get notary parser
+	l.mtx.RLock()
+	parser, ok := l.notaryParsers[notaryKey]
+	l.mtx.RUnlock()
+
+	if !ok {
+		l.log.Debug("notary parser not set")
+
+		return
+	}
+
+	// parse the notary event
+	event, err := parser(notaryEvent.Params())
```

Comment:
there is a special method in the listener that registers parsers and handlers. this line does now execute for removed notary requests since there is no code that registers parsers and listeners for it